oci/oci.go

```diff
@@ -0,0 +1,147 @@
+package oci
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	mutils "github.com/projectatomic/skopeo/manifest"
+	"github.com/projectatomic/skopeo/reference"
+	"github.com/projectatomic/skopeo/types"
+)
+
+type descriptor struct {
+	Digest    string `json:"digest"`
+	MediaType string `json:"mediaType"`
+	Size      int64  `json:"size"`
+}
+
+type ociImageDestination struct {
+	dir string
+	ref reference.Named
+	tag string
+}
+
+// TODO(runcom): remove, it's a copy from docker/docker_utils.go
+// parseDockerImageName converts a string into a reference and tag value.
+func parseDockerImageName(img string) (reference.Named, string, error) {
+	ref, err := reference.ParseNamed(img)
+	if err != nil {
+		return nil, "", err
+	}
+	ref = reference.WithDefaultTag(ref)
+	var tag string
+	switch x := ref.(type) {
+	case reference.Canonical:
+		tag = x.Digest().String()
+	case reference.NamedTagged:
+		tag = x.Tag()
+	}
+	return ref, tag, nil
+}
+
+// NewOCIImageDestination returns an ImageDestination for writing to an existing directory.
+func NewOCIImageDestination(dir string) (types.ImageDestination, error) {
+	// oci dest follow docker's one - parse those here
+	// there's no clear direction on naming though but since image-spec is WIP just assume this for now..
+	ref, tag, err := parseDockerImageName(dir)
```

Comment:
> just find the last : and split into directory + tag?

how do we make sure directory is valid? valid as in `docker.io/fedora` for instance, I think we should require `oci:` prefix to enforce a valid destination, or not?